### ISSUE TITLE
Creating action targeting id in selector not working.

### ISSUE THREAD

**Issue by mrsweaters** [PostHog/posthog#274]
**Describe the bug**
Tried to create an Action targeting an element by id and it is not matching any events:
```html
<input id="startTestingB" type="submit" class="button small-expanded secondary" value="Start Testing">
```
With this in the selector input when creating an action:
```
[id='startTestingB']
```

Also tried targeting by text `Start Testing` but not sure if that works with `value`.

**Comment by mrsweaters** [comment]
Okay, so it looks like it is a sanitization issue:

<img width="987" alt="Screen Shot 2020-03-04 at 1 42 12 PM" src="https://user-images.githubusercontent.com/535897/75925682-f672c280-5e1d-11ea-84b2-f9f02256c510.png">

**Comment by timgl** [comment]
This is a tough one. This definitely isn't correct, however as [mixpanel showed](https://techcrunch.com/2018/02/05/mixpanel-passwords/), making sure you're not accidentally pulling passwords/SSN/CC's is super hard.

I can see 2 solutions:
1. For inputs specifically, only capture attributes we are very confident are safe: `ID`, `name` and `class`
This is what heap.io does too. The downside is, for every other element we're already capturing _all_ attributes, and people are relying on them to do Action matching. This means we'd have to end up explaining "for buttons, you can match against all attributes, for inputs only id, name and class." I think this will be confusing/hard to communicate.
2. Move _everything_ to only capturing 'safe' attributes
This would be consistent, and reduce the volume of data we're storing. However, people are relying on other attributes to do matching at the moment and I'd hate to give up that flexibility.

@mrsweaters Do you have any thoughts on this? I think option 1 makes sense, despite the confusion overhead. 

Doing option 1 would also involve putting 'name' selector back into the Actions creation form.

**Comment by jamesefhawkins** [comment]
Losing the flexibility in (2) would be annoying.

Maybe the best solution to explain (1) to a user by creating an alert in the toolbar only when this happens in practise "you have selected an input, this only works for ID, name, class for security reasons".

Developers represent the vast majority of our users at this time, who I think would understand this, so I'd probably vote there although I'm not 100% convinced.

**Comment by mrsweaters** [comment]
I think approach 1 sounds reasonable. However, it was easy enough for me to switch the input to a button. I'd probably just document it for now and wait until more folks request it.

**Comment by lksv** [comment]
Approach 1 sounds better than 2 for me. I am looking forward to some progress, this blocks create an particular Action for me.

**Comment by timgl** [comment]
@lksv Thanks for bumping this! I've created https://github.com/PostHog/posthog-js/pull/63 to solve this issue

**Comment by mariusandra** [comment]
Great! `1.3.4` released with this in!

**Comment by mariusandra** [comment]
My apologies for the comment above (wrong issue). However we just released version `1.3.4` of posthog-js that should fix this issue. In case you're using the library from npm, you can start using it directly!

**Comment by lksv** [comment]
@mariusandra I can confirm that it works now (at least on hosted app.posthog.com). Thaks!
